plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadataFactory.java

```diff
@@ -58,6 +59,8 @@
     private final boolean deleteSchemaLocationsFallback;
     private final boolean useUniqueTableLocation;
 
+    private final MetastoreTypeConfig metastoreTypeConfig;
```

Comment:
`String metastoreType` is better as other comment.

Reply:
I agree `String metastore` is better in `DeltaLakeMetadata` but in factory I think it is fine to use config object. 
Also it requires less changes as Config is available in guice to inject and String is not ;)